ChangeLog.d/remove-via-padlock-support.txt

```diff
@@ -0,0 +1,4 @@
+Removals
+   * Drop support for VIA Padlock. Removes MBEDTLS_PADLOCK_C.
+     Note that it is still possible to use VIA Padlock through a
+     PSA accelerator driver that is not part of Mbed TLS. Fixes #5903.
```

Comment:
My immediate thought on reading this is: "oh, how do I install this driver?". Would this be better as "Note that this does not prevent users from using VIA Padlock through a suitable PSA driver (not provided by Mbed TLS)"?

Reply:
Happy to change the text here but must admit that I derive exactly the same meaning from both expressions. Not sure how it changes the semantics.

Reply:
"Still possible to use" could be the prelude to a workaround, whereas "does not prevent" doesn't have quite the same implication. I agree that on the surface the semantics of the two statements should be the same, but the former is more open to mis-interpretation

Reply:
Ah, I do see what you mean there actually. Fair cop.

Reply:
I think the changelog entry is still confusing and weirdly emphatic. Why do we mention the possibility of using a driver for such an obscure feature of all things?

On the other hand, we should have a section in the migration guide that explains that if you were using the built-in VIA padlock support, you should write a driver for it. We haven't started the 4.0 migration guide yet, but now is a good time as any.

Reply:
Would you prefer no mention of the driver? I am not sure what is emphatic about mentioning that people could provide their own driver for something we are removing, no matter how obscure the feature is.

Reply:
4.0 will remove a lot of things. I don't think the changelog is the right place to explain how users can live without those things. That's the job of a more detailed document: the migration guide.